Quantum-Control-Applications-QuAM/Superconducting/quam_libs/quam_builder/wiring/create_analog_ports.py

```diff
@@ -0,0 +1,77 @@
+from typing import List
+
+from .paths import OCTAVES_BASE_JSON_PATH, PORTS_BASE_JSON_PATH
+from qualang_tools.wirer.instruments.instrument_channel import AnyInstrumentChannel
+
+def create_octave_port(channel: AnyInstrumentChannel) -> (str, str):
+    """
+    Generates a key/JSON reference pair from which a QuAM port can be created
+    for a single Octave channel.
+    """
+    if channel.io_type == "output":
+        key = "frequency_converter_up"
+    elif channel.io_type == "input":
+        key = "frequency_converter_down"
+    else:
+        raise ValueError(f"Unknown IO type {channel.io_type}")
+
+    # todo: handle "octave" in OPX+ vs "oct" in new QOP?
+    reference = OCTAVES_BASE_JSON_PATH
+    reference += f"/octave{channel.con}"
```

Comment:
This will throw an error with QOP242 for which the Octave name is octx, to avoid this issue, why not specifying the octave_settings `octave_settings = {"oct1": {"ip": "192.168.88.250"} }` or `octave_settings = {"octave1": {"port": 11235} }` under make_wiring instead of make_quam? Then the `build_quam_wiring` function will be able to get the name from there
What do you guys think @deanpoulos @nulinspiratie ?